packages/compiler/server/serverlib.ts

```diff
@@ -465,6 +471,45 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  function getNameandSymbolKind(node: Node): { name: string; kind: SymbolKind } | undefined {
+    switch (node.kind) {
+      case SyntaxKind.NamespaceStatement:
+        return { name: node.id.sv, kind: SymbolKind.Namespace };
+      case SyntaxKind.CadlScript:
+        return { name: node.id.sv, kind: SymbolKind.File };
+      case SyntaxKind.EnumStatement:
+        return { name: node.id.sv, kind: SymbolKind.Enum };
+      case SyntaxKind.InterfaceStatement:
+        return { name: node.id.sv, kind: SymbolKind.Interface };
+      case SyntaxKind.OperationStatement:
+        return { name: node.id.sv, kind: SymbolKind.Operator };
+      default:
+        return undefined;
+    }
+  }
+
+  async function getDocumentSymbols(params: DocumentSymbolParams): Promise<SymbolInformation[]> {
+    const file = await compilerHost.readFile(getPath(params.textDocument));
+    const ast = parse(file);
+    const ranges: SymbolInformation[] = [];
+    visitChildren(ast, addRangesForNode);
+
+    function addRangesForNode(node: Node) {
+      const symbolNode = getNameandSymbolKind(node);
+      if (symbolNode !== undefined) {
+        const start = file.getLineAndCharacterOfPosition(node.pos);
+        const end = file.getLineAndCharacterOfPosition(node.end);
+        ranges.push({
+          name: symbolNode.name,
+          kind: symbolNode.kind,
+          location: Location.create(params.textDocument.uri, Range.create(start, end)),
```

Comment:
Need to add the deprecated tag if a node is deprecated too. Check `addIdentifierCompletion` for example with that, should be similar

Reply:
talked offline and we cannot do that one if we only parse the document. Could be added when we have the workspace symbols.